crap-advisor-frontend/src/core/components/Map/index.tsx

```diff
@@ -0,0 +1,55 @@
+import React, {useEffect, useState} from "react";
+import "./index.css";
+import {MapContainer, TileLayer, ZoomControl} from "react-leaflet";
+import RestroomMarker from "../RestroomMarker";
+import {apiUrl} from "../../../app/constants";
+import {LatLngExpression} from "leaflet";
+import {Restroom} from "../../../app/typings";
+
+export interface IMapProps {
+    selectRestroom: (restroomToSelect: Restroom) => void;
+}
+
+const Map = ({selectRestroom}: IMapProps): JSX.Element => {
+    const defaultPosition: LatLngExpression = [53.21176, 50.18394];
+    const defaultZoom: number = 13;
+    const [restrooms, setRestrooms] = useState<Restroom[]>([]);
+
+    useEffect(() => {
+        try {
+            fetch(apiUrl + "/restroom/all")
```

Comment:
Is there any way to avoid double request at startup? It's really bothering me since the very beginning

Reply:
Im gonna deep dive in this prombem to resolve twice call

Reply:
Good luck on your investigation